src/Aspire.Dashboard/Components/Pages/Metrics.razor.cs

```diff
@@ -182,7 +182,17 @@ private void UpdateApplications()
     {
         _applications = TelemetryRepository.GetApplications();
         _applicationViewModels = ApplicationsSelectHelpers.CreateApplications(_applications);
-        _applicationViewModels.Insert(0, _selectApplication);
+
+        if (_applicationViewModels.Count != 1)
+        {
+            _applicationViewModels.Insert(0, _selectApplication);
+        }
+        else
+        {
+            PageViewModel.SelectedApplication = _applicationViewModels.Single();
+            NavigationManager.NavigateTo(GetUrlFromSerializableViewModel(ConvertViewModelToSerializable()));
```

Comment:
Is this required? The metrics page has already been changed to redirect when there is one app.